site/src/i18n/en/workspacePage.json

```diff
@@ -6,5 +6,21 @@
     "schedule": "Schedule",
     "editDeadlineMinus": "Subtract one hour",
     "editDeadlinePlus": "Add one hour"
+  },
+  "actionButton": {
+    "start": "Start",
+    "stop": "Stop",
+    "delete": "Delete",
+    "cancel": "Cancel",
+    "update": "Update",
+    "updating": "Updating",
+    "starting": "Starting...",
+    "stopping": "Stopping...",
+    "deleting": "Deleting..."
+  },
```

Comment:
Same sort of comment here: I like linking the language to the actual typing that's coming back from the API - that way we get a 1:1 mapping of BE state and UI. In the past, I've managed with constants like:
````
export const ActionButtonLabels = {
  [WorkspaceStatus.starting]: i18n.t('workspacePage:actionButton.starting'),
  [WorkspaceStatus.stopping]: i18n.t('workspacePage:actionButton.stopping'),
  etc...
} as const;
````

But that was with enums. @presleyp or @jsjoeio again, is there a way to do this with literals? I feel like I could come over to the dark side if there is.

Reply:
I think we could define this but with "starting" instead of `[WorkspaceStatus.starting]`, which is not as clear that it's typesafe, but is typesafe. And then we'd know that every value of `WorkspaceStatus` mapped to some text. How's that? (@jsjoeio def interested in your input on all of this when you're available!)

Reply:
That would be nice, if you feel it's a good add. Not a dealbreaker if you don't! 

Also yeah, so curious to hear if there's a way to directly index.

Reply:
I ended up just changing the way I handle text for the disabled buttons, so that it's not a dynamic lookup.

Reply:
I'm not sure how `i18n` could let us pull in TypeScript types 🤔 Ahhh I see @Kira-Pilot's comment above. I think that's a great idea! Happy to look into this after this PR is merged too.